app/components/Views/ImportNewSecretRecoveryPhrase/index.tsx

```diff
@@ -0,0 +1,437 @@
+import Text, {
+  TextVariant,
+} from '../../../component-library/components/Texts/Text';
+import Icon, {
+  IconSize,
+  IconName,
+} from '../../../component-library/components/Icons/Icon';
+import ButtonLink from '../../../component-library/components/Buttons/Button/variants/ButtonLink';
+import { ButtonSize } from '../../../component-library/components/Buttons/Button';
+import { wordlist } from '@metamask/scure-bip39/dist/wordlists/english';
+import { isValidMnemonic } from '../../../util/validators';
+import useCopyClipboard from '../Notifications/Details/hooks/useCopyClipboard';
+import ClipboardManager from '../../../core/ClipboardManager';
+import BannerAlert from '../../../component-library/components/Banners/Banner/variants/BannerAlert';
+import { BannerAlertSeverity } from '../../../component-library/components/Banners/Banner';
+import SelectComponent from '../../UI/SelectComponent';
+import {
+  ToastContext,
+  ToastVariants,
+} from '../../../component-library/components/Toast';
+import { useSelector } from 'react-redux';
+import { selectHDKeyrings } from '../../../selectors/keyringController';
+
+const defaultNumberOfWords = 12;
+
+export const parseSecretRecoveryPhrase = (seedPhrase: string) =>
+  (seedPhrase || '').trim().toLowerCase().match(/\w+/gu)?.join(' ') || '';
+
+const srpOptions = [
+  {
+    value: 12,
+    label: strings('import_new_secret_recovery_phrase.12_word_option'),
+  },
+  {
+    value: 24,
+    label: strings('import_new_secret_recovery_phrase.24_word_option'),
+  },
+];
+
+/**
+ * View that's displayed when the user is trying to import a new secret recovery phrase
+ */
+const ImportNewSecretRecoveryPhrase = () => {
+  const [inputWidth, setInputWidth] = useState<DimensionValue | undefined>(
+    Device.isAndroid() ? '99%' : undefined,
+  );
+  const navigation = useNavigation();
+  const mounted = useRef<boolean>(false);
+  const { colors, themeAppearance } = useAppTheme();
+  const styles = createStyles(colors);
+  const { toastRef } = useContext(ToastContext);
+
+  const [numberOfWords, setNumberOfWords] = useState(defaultNumberOfWords);
+  const [secretRecoveryPhrase, setSecretRecoveryPhrase] = useState<string[]>(
+    Array(numberOfWords).fill(''),
+  );
+  const [showPassword, setShowPassword] = useState(true);
+  const [loading, setLoading] = useState(false);
+  const [srpError, setSrpError] = useState('');
+  const [invalidSRPWords, setInvalidSRPWords] = useState<boolean[]>(
+    Array(numberOfWords).fill(false),
+  );
+  const hdKeyrings = useSelector(selectHDKeyrings);
+
+  const copyToClipboard = useCopyClipboard();
+
+  useEffect(() => {
+    mounted.current = true;
+    // Workaround https://github.com/facebook/react-native/issues/9958
+    if (inputWidth) {
+      setTimeout(() => {
+        mounted.current && setInputWidth('100%');
+      }, 100);
+
+      return () => {
+        mounted.current = false;
+      };
+    }
+    // eslint-disable-next-line
+  }, []);
```

Comment:
S: Be careful of this effect cleanup in case of unmount!
If the component unmounts before `setTimeout` executes, `mounted.current = false;` inside the cleanup function will not prevent `setInputWidth('100%')` from running.

```suggestion
  useEffect(() => {
    mounted.current = true;
    // Workaround https://github.com/facebook/react-native/issues/9958
    if (inputWidth) {
      const timeoutId = setTimeout(() => {
        mounted.current && setInputWidth('100%');
      }, 100);

      return () => {
        clearTimeout(timeoutId);
        mounted.current = false;
      };
    }
    // eslint-disable-next-line
  }, []);
```